R/PipeOpLearner.R

```diff
@@ -79,6 +79,12 @@
 PipeOpLearner = R6Class("PipeOpLearner", inherit = PipeOp,
   public = list(
     initialize = function(learner, id = if (is.character(learner)) learner else learner$id, param_vals = list()) {
```

Comment:
I think the solution here would be to have `id = NULL`, and then in the code
```r
private$.learner = as_learner(learner, clone = TRUE)  # merge master before doing this!
self$id = id %??% private$.learner$id
```
also add a tests pls